src/main.rs

```diff
@@ -79,74 +81,90 @@ fn main() -> Result<(), Box<dyn Error>> {
     info!("start");
 
     let config = Config::from_args();
-    let (send, recv) = sync_channel(config.channels_size.into());
-    let handle = blocks_iterator::iterate(config, send);
-    let mut period = PeriodCounter::new(Duration::from_secs(10));
-
-    while let Some(block_extra) = recv.recv()? {
-        if period.period_elapsed().is_some() {
-            info!(
-                "# {:7} {}",
-                block_extra.height,
-                block_extra.block_hash,
-            );
-        }
-
-        // First block of 2020
-        if block_extra.height < 610682 {
-        // if block_extra.height < 100 {
-            continue;
-        }
 
-        for tx in block_extra.block.txdata {
-            for input in tx.input {
-                fn check<Ctx: ScriptContext, Ctx2: miniscript_new::ScriptContext>(s: &Script, prev_out: &bitcoin::OutPoint) {
-                    match check_script::<Ctx, Ctx2>(s) {
-                        CheckRes::MaybeVuln => warn!("Found potentially vulnerable input: {}", prev_out),
-                        CheckRes::Vuln => warn!("Found vulnerable input: {}", prev_out),
-                        CheckRes::Safe => {},
+    blocks_iterator::iter(config)
+        .flat_map(|mut block_extra| {
+            if block_extra.height < 1_000_000 {
+                return Vec::new();
+            }
+            let mut vec = vec![];
```

Comment:
I think there's a way to avoid allocating the vec entirely, I'm not sure how beneficial it could be

Reply:
surely better :) patch applied